File: competitive_programming/codeforces/11_golang/546a_soldier_and_bananas.go
package main

import "fmt"

func soldierAndBananas(cost, money, count int64) int64 {
	total := cost * count * (count + 1) / 2
	if total > money {
		return total - money
	}
	return 0
}

func main() {
	if soldierAndBananas(3, 17, 4) != 13 || soldierAndBananas(1, 100, 1) != 0 {
		panic("soldier and bananas failed")
	}
	fmt.Println("546A soldier and bananas ok")
}

File: competitive_programming/codeforces/07_typescript/546a_soldier_and_bananas.ts
function soldierAndBananas(cost: number, money: number, count: number): number {
  const total = (cost * count * (count + 1)) / 2;
  return Math.max(0, total - money);
}

if (soldierAndBananas(3, 17, 4) !== 13) throw new Error("soldier and bananas failed");
if (soldierAndBananas(1, 100, 1) !== 0) throw new Error("soldier and bananas failed");
console.log("546A soldier and bananas ok");
